src/test/java/com/rakovets/course/java/core/practice/generics/obscure/ObscureTest.java

```diff
@@ -0,0 +1,91 @@
+package com.rakovets.course.java.core.practice.generics.obscure;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ObscureTest {
+    @Test
+    void isEmptyTest1() {
+        Obscure<String> obscureEmpty = new Obscure<>(null);
+
+        Assertions.assertTrue(obscureEmpty.isEmpty());
+    }
+
+    @Test
+    void isEmptyTest2() {
+        Obscure<String> obscure = new Obscure<>("Obscure");
+
+        Assertions.assertFalse(obscure.isEmpty());
+    }
+
+    @Test
+    void isPresentTest1() {
+        Obscure<String> obscure = new Obscure<>(null);
+
+        Assertions.assertFalse(obscure.isPresent());
+    }
+
+    @Test
+    void isPresentTest2() {
+        Obscure<String> obscure = new Obscure<>("Obscure");
+
+        Assertions.assertTrue(obscure.isPresent());
+    }
+
+    @Test
+    void orElseTest1() {
+        Obscure<Object> obscureNotPresent = new Obscure<>(null);
+        Double object = 2.0;
+
+        Object result = obscureNotPresent.orElse(object);
+        Double expected = 2.0;
+
+        Assertions.assertEquals(expected, result);
```

Comment:
```suggestion

        Assertions.assertEquals(2.0, result);
```

another places - alike